src/internal/modules/billing/controllers/bill-run.js

```diff
@@ -257,6 +252,49 @@ const getBillingBatchEmpty = async (request, h) => {
   })
 }
 
+/**
+ * Renders an error page if the batch is errored - i.e. the charging module is down
+ * @param {Object} request.pre.batch - billing batch loaded by pre handler
+ */
+const getBillingBatchError = async (request, h) => {
+  const { batch } = request.pre
+
+  return h.view('nunjucks/billing/batch-error', {
+    ...request.view,
+    pageTitle: getBillRunPageTitle(batch),
+    batch,
+    back: BATCH_LIST_ROUTE,
+    errorList: _errorList(batch.errorCode)
+  })
+}
+
+const _errorList = (errorCode) => {
+  if (!errorCode) {
+    return [{ text: 'No error code was assigned. We have no further information at this time.' }]
+  }
+
+  const errors = [
+    { code: 10, text: 'Error when populating the charge versions.' },
+    { code: 20, text: 'Error when processing the charge versions.' },
+    { code: 30, text: 'Error when preparing the transactions.' },
+    { code: 40, text: 'Error when requesting or processing a transaction charge.' },
+    { code: 50, text: 'Error when creating the Charging Module bill run.' },
+    { code: 60, text: 'Error when deleting an invoice.' },
+    { code: 70, text: 'Error when processing two-part tariff.' },
+    { code: 80, text: 'Error when getting the Charging Module bill run summary.' },
+    { code: 90, text: 'Error when re-billing a bill run.' }
+  ]
+
+  const error = errors.find((error) => {
+    if (error.code === errorCode) {
+      return true
+    }
+    return false
+  })
+
+  return [{ text: error.text }]
```

Comment:
Are we confident that if there's an error code, it will always be one that's in the list of `errors`? Or do we want to add a little additional checking to return something like `[{ text: 'An unknown error code was assigned. We have no further information at this time.' }]`

Reply:
I thought about that, and TBH I want it to explode should this happen. Else I fear we'd run the risk of not exposing the fact we'd missed this should we add a new error code.

Regards the list of codes, they are all defined in the [water-abstraction-service](https://github.com/DEFRA/water-abstraction-service/blob/main/src/lib/models/batch.js#L33)